### PR TITLE
CLOUDSTACK-10168 - VR duplicate entries in /etc/hosts when reusing VM name

### PR DESCRIPTION
**Pull request by bwsw**
Bug is fixed. Fixed typo 'adress' to 'address' everywhere which leads to correct generation for agent communication json dhcp command.

@rhtyd
@dagsonstebo

Take a look please. Fixes the bug reported by Dag Sonstebo for Basic zone but unable to test carefully for Advanced zone VR. 

Tested in our pre-prod 4.10. 

Create / abc (x.y.z.11), cde (x.y.z.13)
Remove / abc, cde
Create / cde -> /etc/hosts (x.y.z.11 -> cde, x.y.z.13 removed)

### CODE REVIEW

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test

File: systemvm/patches/debian/config/opt/cloud/bin/cs_dhcp.py [hunk 1]
```diff
@@ -18,17 +18,22 @@
...
-        return dbag
+    if "add" in data and data['add'] is False and "ipv4_address" in data:
+        if data['ipv4_address'] in dbag:
+            del(dbag[data['ipv4_address']])
     else:
-        dbag[data['ipv4_adress']] = data
-    return dbag
+        remove_key = None
+        for key, entry in dbag.iteritems():
+            if key <> 'id' and entry['host_name'] == data['host_name']:
```
Comment (rhtyd) [inline systemvm/patches/debian/config/opt/cloud/bin/cs_dhcp.py]: `<>` is deprecated and removed in Python 3.x, please use `!=` instead.

**Comment by bwsw** [comment]
@rhtyd code is fixed ( <> changed to != )

**Comment by bwsw** [comment]
@rhtyd 
@wido 
Also, I haven't found if IPv6 dns resolution is implemented in VR dnsmasq currently, at least ipv6 records are not written into /etc/hosts. Can you clarify?

**Comment by rhtyd** [comment]
Tests LGTM, merging based on two lgtms and results (ignoring known failures).

**Pull request merged**